src/pu-package.c

```diff
@@ -0,0 +1,253 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright (c) 2023 PHYTEC Messtechnik GmbH
+ */
+
+#include <fcntl.h>
+#include <gio/gio.h>
+#include <glib.h>
+#include <glib/gstdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include "pu-error.h"
+#include "pu-package.h"
+#include "pu-utils.h"
+
+G_DEFINE_QUARK(pu-package-context-error-quark, pu_package_error)
+
+gchar *
+pu_package_get_layout_file(const gchar *pwd,
+                           GError **error)
+{
+    g_autofree gchar *layout_file = NULL;
+
+    layout_file = g_build_filename(pwd, PU_PACKAGE_LAYOUT_BASENAME, NULL);
+    if (!g_file_test(layout_file, G_FILE_TEST_IS_REGULAR)) {
+        g_set_error(error, PU_PACKAGE_ERROR, PU_PACKAGE_ERROR_MISSING_LAYOUT,
+                    "No layout file '%s' found", g_path_get_basename(layout_file));
+        return NULL;
+    }
+
+    return g_steal_pointer(&layout_file);
+}
+
+gboolean
+pu_package_create(gchar **files,
+                  const gchar *output,
+                  gboolean force_overwrite,
+                  GError **error)
+{
+    g_autofree gchar *cmd = NULL;
+    g_autofree gchar *input = NULL;
+    gboolean has_layout_yaml = FALSE;
+
+    g_return_val_if_fail(files != NULL, FALSE);
+    g_return_val_if_fail(g_strcmp0(output, "") > 0, FALSE);
+    g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+    /* Check if specified input files exist */
+    for (guint i = 0; i < g_strv_length(files); i++) {
+        g_autofree gchar *file = g_strdup(files[i]);
+        if (!g_file_test(file, G_FILE_TEST_EXISTS)) {
+            g_set_error(error, PU_PACKAGE_ERROR, PU_PACKAGE_ERROR_NOT_FOUND,
+                        "Input file '%s' does not exist", file);
+            return FALSE;
+        }
+    }
+
+    /* Check if input files contain required layout configuration */
+    for (guint i = 0; i < g_strv_length(files); i++) {
+        g_autofree gchar *file = g_strdup(files[i]);
+        if (g_str_equal(file, "layout.yaml"))
+            has_layout_yaml = TRUE;
+    }
```

Comment:
Combine the two loops as they iterate over the same array.

Reply:
Done.